spec/integration/knife/client_bulk_delete_spec.rb

```diff
@@ -0,0 +1,109 @@
+#
+# Copyright:: Copyright 2013-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "support/shared/integration/integration_helper"
+require "support/shared/context/config"
+
+describe "knife client bulk delete", :workstation do
+  include IntegrationSupport
+  include KnifeSupport
+
+  include_context "default config options"
+
+  when_the_chef_server "has some clients" do
+    before do
+      client "cons", {}
+      client "car", {}
+      client "cdr", {}
+      client "cat", {}
+    end
+
+    it "deletes all matching clients" do
+      knife("client bulk delete ^ca.*", input: "Y").should_succeed <<EOM
```

Comment:
Reading this test, my primary question is "what happens if I don't specify the ^ anchor.